aws-iot-mitigationaction/src/main/java/com/amazonaws/iot/mitigationaction/Translator.java

```diff
@@ -0,0 +1,201 @@
+package com.amazonaws.iot.mitigationaction;
+
+import software.amazon.awssdk.services.iot.model.InternalFailureException;
+import software.amazon.awssdk.services.iot.model.InvalidRequestException;
+import software.amazon.awssdk.services.iot.model.IotException;
+import software.amazon.awssdk.services.iot.model.LimitExceededException;
+import software.amazon.awssdk.services.iot.model.MitigationActionParams;
+import software.amazon.awssdk.services.iot.model.ResourceAlreadyExistsException;
+import software.amazon.awssdk.services.iot.model.ResourceNotFoundException;
+import software.amazon.awssdk.services.iot.model.Tag;
+import software.amazon.awssdk.services.iot.model.ThrottlingException;
+import software.amazon.awssdk.services.iot.model.UnauthorizedException;
+import software.amazon.cloudformation.exceptions.BaseHandlerException;
+import software.amazon.cloudformation.exceptions.CfnAccessDeniedException;
+import software.amazon.cloudformation.exceptions.CfnAlreadyExistsException;
+import software.amazon.cloudformation.exceptions.CfnInternalFailureException;
+import software.amazon.cloudformation.exceptions.CfnInvalidRequestException;
+import software.amazon.cloudformation.exceptions.CfnNotFoundException;
+import software.amazon.cloudformation.exceptions.CfnServiceLimitExceededException;
+import software.amazon.cloudformation.exceptions.CfnThrottlingException;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * This class is a centralized placeholder for
+ *  - api request construction
+ *  - object translation to/from aws sdk
+ *  - resource model construction for read/list handlers
+ */
+public class Translator {
+
+
+  static BaseHandlerException translateIotExceptionToCfn(IotException e) {
+
+    // We're handling all the exceptions documented in API docs
+    // https://docs.aws.amazon.com/iot/latest/apireference/API_CreateMitigationAction.html#API_CreateMitigationAction_Errors (+same pages for other APIs)
+    // For Throttling and InternalFailure, we want CFN to retry, and it will do so based on the exception type.
+    // Reference with Retriable/Terminal in comments for each: https://tinyurl.com/y378qdno
+    if (e instanceof ResourceAlreadyExistsException) {
+      // Note regarding idempotency:
+      // CreateMitigationAction API allows tags. CFN attaches its own stack level tags with the request. If a MitigationAction
+      // is created out of band and then the same request is sent via CFN, API will throw RAEE because the CFN request will have,
+      // extra stack level tags. This behavior satisfies the CreateHandler contract.
+      return new CfnAlreadyExistsException(e);
+    } else if (e instanceof InvalidRequestException) {
+      return new CfnInvalidRequestException(e);
+    } else if (e instanceof LimitExceededException) {
+      return new CfnServiceLimitExceededException(e);
```

Comment:
Minor: somehow the indentations in this file are size 2 instead of 4 like everywhere else

Reply:
Yeah. It is weird. Auto indentation did not work either.
Fixed it with this(In case we face again): "Settings | Editor | Code Style" -- try disabling "Detect and use existing file indents for editing" in case if you have it enabled (it is by default). NOTE: re-opening file in editor may be required.